homeassistant/components/mqtt/__init__.py

```diff
@@ -956,10 +958,11 @@ def _mqtt_on_disconnect(self, _mqttc, _userdata, result_code: int) -> None:
 
     async def _wait_for_mid(self, mid):
         """Wait for ACK from broker."""
-        if mid not in self._pending_operations:
-            self._pending_operations[mid] = asyncio.Event()
+        # Create the mid event if not created, either _mqtt_handle_mid or _wait_for_mid
+        # may be executed first.
+        event = self._pending_operations.setdefault(mid, asyncio.Event())
```

Comment:
I don't think that this line is an improvement -> you're now creating and throwing away a class if the key exists